docs/rst/manual/_ext/ert_jobs.py

```diff
@@ -0,0 +1,198 @@
+
+    def _create_job_node(self):
+        job_section_node = nodes.section(ids=[self.name])
+        title_node = nodes.title(text=self.name)
+        job_section_node.append(title_node)
+        return job_section_node
+
+    def _add_job_config_section(self, node):
+        config_section_node = nodes.section(ids=[self.name + "-job-config"])
+        job_config_title_node = nodes.title(text="ERT config file")
+        with open(self.job_config_file) as fh:
+            job_config_text = fh.read()
+        job_config_text_node = nodes.literal_block(text=job_config_text)
+        config_section_node.append(job_config_title_node)
+        config_section_node.append(job_config_text_node)
+        node.append(config_section_node)
+
+    def _add_job_details(self, node):
+        job_namespace_node = nodes.definition_list_item(
+            "",
+            nodes.term("", "", nodes.strong(text="Namespace")),
+            nodes.definition("", nodes.strong(text=self.namespace)),
+        )
+        job_package_node = nodes.definition_list_item(
+            "",
+            nodes.term("", "", nodes.strong(text="Source package")),
+            nodes.definition("", nodes.strong(text=self.job_source)),
+        )
+        job_details_node = nodes.definition_list(
+            "", job_namespace_node, job_package_node
+        )
+        node.append(job_details_node)
+
+    def _add_job_description_section(self, node, state):
+        _parse_raw_rst(self.description, node, state)
+        self._add_job_details(node)
+
+    def _add_example_section(self, node, state):
+        example_title_node = nodes.title(text="Examples")
+
+        example_section_node = nodes.section(ids=[self.name + "-examples"])
+        example_section_node.append(example_title_node)
+
+        _parse_raw_rst(self.examples, example_section_node, state)
+
+        node.append(example_section_node)
+
+    def _add_argparse_section(self, node, state):
+        argparse_title_node = nodes.title(text="Script arguments")
+
+        argparse_section_node = nodes.section(ids=[self.name + "-argparse"])
+        argparse_section_node.append(argparse_title_node)
+
+        _parse_raw_rst(argparse_template_text, argparse_section_node, state)
+
+        node.append(argparse_section_node)
+
+    def add_to_node(self, root_node, state):
+        node = self._create_job_node()
+        self._add_job_description_section(node, state)
+        self._add_job_config_section(node)
+        if self.argparse is not None:
+            self._add_argparse_section(node, state)
+        if self.examples:
+            self._add_example_section(node, state)
+        root_node.append(node)
+
+
+class ErtJobs(SphinxDirective):
+
+    has_content = True
+
+    def _generate_namespace_sections(self, jobs):
+
+        sections = dict()
+        for job_name, docs in jobs.items():
+            if job_name.islower():
+                continue
+
+            namespace = docs.get("namespace", "other")
```

Comment:
Magic strings could be defined once in this module.

Reply:
Added them as static variables to the class, or do you prefer to have them in the module?